com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventData.cs

```diff
@@ -323,6 +341,32 @@ private int SortNetworkObjects(NetworkObject first, NetworkObject second)
             return 0;
         }
 
+        /// <summary>
```

Comment:
This feels like scope creep. Don't we already have deferred parenting support? And doesn't this not correctly handle sorting a hierarchy nesting level > 1?

Reply:
If you enable developer logging and run the associated integration test you will see that it nests 10 children (each child is a parent and has a child) and is spawned in reverse order to make children spawned (server-side) before their respective parent.  When clients are connecting and synchronizing, the final synchronization serialization order yields the parents before the children (i.e. it sorts a hierarchy nesting level > 1).

It only sorts during synchronization to help assure that a child will always have a parent available.
When dynamically spawning and parenting, then the orphan child list is used.

Reply:
Sorting them for client synchronization helps when there are multi-generation nested children that all have varying scales.  We want to assure that it applies from the root parent down to the last nested child in regards to ordering.

Reply:
Could this be handled using DeferredMessageManager?

If a spawn is received for an object that has a parent set, but the parent doesn't exist, CreateObjectMessage can do this:

```csharp

if (ObjectInfo.HasParent && !networkManager.SpawnManager.SpawnedObjects.ContainsKey(ObjectInfo.ParentObjectId))
{
    networkManager.DeferredMessageManager.DeferMessage(IDeferredMessageManager.TriggerType.OnSpawn, ObjectInfo.ParentObjectId, reader, ref context);
    return false;
}
```

I think that'd handle this case and also other potential edge cases with less complexity.

Reply:
(The way that works, if it's not obvious, is that on the receive side, if it receives the messages in the wrong order, it defers them with a specific trigger. The OnSpawn trigger means that, when an object with a NetworkObjectId matching the second parameter is spawned, the deferred message is processed immediately... so basically it just does this same sorting operation for you, it just does it on the receiving end in a generic way.)

Reply:
This is only for initial client synchronization.  So there is one really large message that has all NetworkObjects in it.  We can remove this, but it does remove the chance of a child being added to the orphaned children list during the initial synchronization and if user code depends upon a parent existing when a child is spawned the parent is guaranteed to have already been spawned and parenting by the time OnNetworkSpawn is invoked on the child.